UI: Update deprecated Icons with the new @storybook/icons in addons

cdedreuille opened the pull request:
Closes https://github.com/storybookjs/storybook/issues/25760

## What I did

As part of Storybook 8 we are transitioning to a new package for icons. In this PR we are transforming the `<Icons />` component into the new icons from `@storybook/icons`.

## Checklist for Contributors

### Testing

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to communicate how to test your changes -->

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

<!-- Please include the steps to test your changes here. For example:

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access X story

-->

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

cdedreuille committed:
Remove Icons component

cdedreuille committed:
Update ToolbarMenuButton.tsx

cdedreuille committed:
Update preview.tsx

cdedreuille committed:
Update types

cdedreuille committed:
Merge branch 'next' into charles-fix-toolbars-icons

cdedreuille committed:
Bring back old version

cdedreuille committed:
Remove deprecate warning

cdedreuille committed:
Replace icons

cdedreuille committed:
Continue to remove Icons

cdedreuille committed:
Remove other icons from addons

cdedreuille committed:
Merge branch 'next' into charles-fix-toolbars-icons

cdedreuille committed:
Update shortcuts.tsx

cdedreuille committed:
Update Color.tsx

cdedreuille committed:
Update SectionRow.tsx

JReinhold commented on code/addons/interactions/src/components/StatusIcon.tsx:
```diff
@@ -1,51 +1,35 @@
-      status={status}
-      icon={icon}
-      className={className}
-    />
+    <>
+      {status === CallStates.DONE && <CheckIcon color={theme.color.positive} />}
+      {status === CallStates.ERROR && <StopAltIcon color={theme.color.negative} />}
+      {status === CallStates.ACTIVE && <PlayIcon color={theme.color.secondary} />}
+      {status === CallStates.WAITING && (
+        <WarningContainer>
+          <CircleIcon color={transparentize(0.5, '#CCCCCC')} size={6} />
+        </WarningContainer>
+      )}
+    </>
   );
```
I feel like this would be much cleaner with a switch statement, something like

```suggestion
  switch (status) {
	  case CallStates.DONE: {
	    return <CheckIcon color={theme.color.positive} />;
	  }
	  case CallStates.ERROR: {
	    return <StopAltIcon color={theme.color.negative} />;
	  }
	  case CallStates.ACTIVE: {
	    return <PlayIcon color={theme.color.secondary} />;
	  }
	  case CallStates.WAITING: {
	    return (
	      <WarningContainer>
	        <CircleIcon color={transparentize(0.5, '#CCCCCC')} size={6} />
	      </WarningContainer>
	    );
	  }
	  default: {
	    return null; //log warning?
	  }
  }

```

cdedreuille replied:
Thanks @JReinhold. Much cleaner indeed

JReinhold commented on code/addons/jest/package.json:
```diff
@@ -54,6 +54,7 @@
   },
   "dependencies": {
     "@storybook/global": "^5.0.0",
+    "@storybook/icons": "^1.2.3",
```
needs to be in `devDependencies`

cdedreuille replied:
Done

JReinhold commented on code/ui/components/src/components/icon/icon.tsx:
```diff
@@ -24,18 +24,27 @@ export interface IconsProps extends ComponentProps<typeof Svg> {
+export const Icons = ({
+  icon,
+  useSymbol,
+  removeDeprecateWarning = false,
```
I'd suggest `__suppressDeprecationWarning` instead.

1. the `__` prefix highlights that this is a prop only intended for internal use
2. `remove` is a destructive action, we're not really removing it everywhere, we're just suppressing it here.

cdedreuille replied:
Done

JReinhold commented on docs/addons/writing-addons.md:
````diff
@@ -98,10 +98,11 @@ Going through the code blocks in sequence:
 ```
 
-The [`useGlobals`](./addons-api.md#useglobals) and [`useStorybookApi`](./addons-api.md#usestorybookapi) hooks from the `manager-api` package are used to access the Storybook's APIs, allowing users to interact with the addon, such as enabling or disabling it. The `Icons` and `IconButtons` components from the [`@storybook/components`](https://www.npmjs.com/package/@storybook/components) package ren ...
+The [`useGlobals`](./addons-api.md#useglobals) and [`useStorybookApi`](./addons-api.md#usestorybookapi) hooks from the `manager-api` package are used to access the Storybook's APIs, allowing users to interact with the addon, such as enabling or disabling it. The `IconButton` component from the [`@storybook/components`](https://www.npmjs.com/package/@storybook/components) package render the button ...
````
@kylegach input here?

kylegach replied:
There's significant changes needed here. I'll push a commit in a bit.

kylegach replied:
@cdedreuille — I pushed [the commit](https://github.com/storybookjs/storybook/pull/25822/commits/997fe1161a19b6bececf4b4b2a2caa86fffd32ed) with further updates.

cdedreuille replied:
Thanks Kyle. Looks great!

JReinhold reviewed (commented):
code looks great! 🔥 I haven't tested the functionality yet though, I will soon.

cdedreuille committed:
Update StatusIcon.tsx

cdedreuille committed:
Update Storybook icons package

cdedreuille committed:
Update props name

cdedreuille committed:
Merge branch 'next' into charles-fix-toolbars-icons

cdedreuille committed:
Update StatusIcon.tsx

kylegach committed:
Further docs updates

- Clarify prose
- Update all remaining usages of `Icons` component to `@storybook/icons`

kylegach commented on docs/addons/writing-addons.md:
````diff
@@ -97,11 +97,13 @@ Going through the code blocks in sequence:
 ```
 
-The [`useGlobals`](./addons-api.md#useglobals) and [`useStorybookApi`](./addons-api.md#usestorybookapi) hooks from the `manager-api` package are used to access the Storybook's APIs, allowing users to interact with the addon, such as enabling or disabling it. The `Icons` and `IconButtons` components from the [`@storybook/components`](https://www.npmjs.com/package/@storybook/components) package ren ...
+The [`useGlobals`](./addons-api.md#useglobals) and [`useStorybookApi`](./addons-api.md#usestorybookapi) hooks from the `manager-api` package are used to access the Storybook's APIs, allowing users to interact with the addon, such as enabling or disabling it.
````
@cdedreuille — I pushed [the commit](https://github.com/storybookjs/storybook/pull/25822/commits/997fe1161a19b6bececf4b4b2a2caa86fffd32ed) with further updates.

kylegach committed:
Format snippet

cdedreuille committed:
Merge branch 'next' into charles-fix-toolbars-icons

cdedreuille committed:
Update icon library
